### PR TITLE
[menu-bar][cli] Add basic support for launching Expo updates

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

Expo Updates are a big part of EAS and Orbit should be able to directly launch them into devices and simulators.

Closes ENG-11072
Closes https://github.com/expo/orbit/issues/113

# How

- Add new `launch-update` command 
- Add @expo/multipart-body-parser and expo-manifests dependencies 
- Add `Emulator.getRunningDeviceAsync` and `Simulator.isSimulatorAsync` helper functions
- Refactor deeplink parsing on menu-bar

### Officially supported formats
This PR adds basic support for launching Expo updates that use the following formats:
- `expo-orbit//update/u.expo.dev/:appId/channel/:channelId?platform=(android|ios)`
- `expo-orbit//update/u.expo.dev/:appId/branch/:branchId?platform=(android|ios)`
- `expo-orbit//update/u.expo.dev/:appId/group/:groupId?platform=(android|ios)`
- `expo-orbit//update/u.expo.dev/:appId/update/:updateId?platform=(android|ios)`

> [!NOTE]  
> As long as the URL specifies the “platform” in the query params Orbit should be able to launch any update.

# Test Plan

### Scenario 1 (An app that uses dev-client)

1. Create development builds on EAS for Android and iOS
2. Create a new update 
3. Launch Orbit using the update URL 
3.1. Most recent development build for that runtime version should be installed automatically 
3.2. Update should be launched 

### Scenario 2 (An app that does not use dev-client)

1. Create a new update (`runtimeVersion=exposdk:X.X.X`)
2. Launch Orbit using the update URL 
2.1. Expo Go for that sdk should be installed automatically 
2.2. Update should be launched

### CODE REVIEW

File: apps/cli/src/commands/LaunchUpdate.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+function getUpdateDeeplink(updateURL: string, manifest: Manifest) {
+  const updateIdURL = updateURL.startsWith('https://u.expo.dev')
+    ? `https://u.expo.dev/update/${manifest.id}`
+    : updateURL;
+
+  const scheme = Array.isArray(manifest?.extra?.expoClient?.scheme)
+    ? manifest?.extra?.expoClient?.scheme[0]
+    : manifest?.extra?.expoClient?.scheme;
+  scheme;
```
Comment (alanjhughes) [inline apps/cli/src/commands/LaunchUpdate.ts:128]:
```suggestion
```

File: apps/cli/src/commands/LaunchUpdate.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    appId: manifest.extra?.eas?.projectId ?? '',
+    platform,
+    distribution,
+  });
+
+  const build = app?.byId?.buildsPaginated?.edges?.[0]?.node;
+  if (
+    build.__typename === 'Build' &&
+    build.expirationDate &&
+    new Date(build.expirationDate) > new Date() &&
```
Comment (alanjhughes) [inline apps/cli/src/commands/LaunchUpdate.ts:184]:
```suggestion
    new Date(build.expirationDate) > Date.now() &&
```
Not a problem at all, I just think it's clearer but up to you

File: apps/menu-bar/src/popover/Core.tsx [hunk 3]
```diff
@@ -129,7 +135,33 @@ function Core(props: Props) {
...
 
-  // @TODO: create another hook
+  const getDeviceByPlatform = useCallback(
+    (platform: 'android' | 'ios') => {
+      const selectedDevicesId = selectedDevicesIds[platform];
+      if (selectedDevicesId && devicesPerPlatform[platform].devices.has(selectedDevicesId)) {
+        return devicesPerPlatform[platform].devices.get(selectedDevicesId);
+      }
+
+      const devices = devicesPerPlatform[platform].devices.values();
```
Comment (alanjhughes) [inline apps/menu-bar/src/popover/Core.tsx:145]:
```suggestion
      const devices = devicesPerPlatform[platform]
      const selectedDevicesId = selectedDevicesIds[platform];
      if (selectedDevicesId && devices.has(selectedDevicesId)) {
        return devices.get(selectedDevicesId);
      }
```
We access `devicesPerPlatform[platform]` a few times, maybe make it variable?

File: apps/menu-bar/src/popover/Core.tsx [hunk 4]
```diff
@@ -129,7 +135,33 @@ function Core(props: Props) {
...
+
+      for (const device of devices) {
```
Comment (alanjhughes) [inline apps/menu-bar/src/popover/Core.tsx:147]:
```suggestion
      for (const device of devices.values()) {
```

File: apps/menu-bar/src/popover/Core.tsx [hunk 5]
```diff
@@ -129,7 +135,33 @@ function Core(props: Props) {
...
+        if (isVirtualDevice(device) && device.state === 'Booted') {
+          setSelectedDevicesIds((prev) => ({ ...prev, [platform]: getDeviceId(device) }));
+          return device;
+        }
+      }
+
+      const [firstDevice] = devicesPerPlatform[platform].devices.values();
```
Comment (alanjhughes) [inline apps/menu-bar/src/popover/Core.tsx:154]:
```suggestion
      const [firstDevice] = devices.values();
```

**Review by alanjhughes — APPROVED** [review]
LGTM. Left a few nits but totally up to you if you want to do anything about them or not

File: apps/cli/src/commands/LaunchUpdate.ts (hunk 1 above)
Comment (gabrieldonadel) [reply apps/cli/src/commands/LaunchUpdate.ts:128]: opss

File: apps/cli/src/commands/LaunchUpdate.ts (hunk 2 above)
Comment (gabrieldonadel) [reply apps/cli/src/commands/LaunchUpdate.ts:184]: sure, makes sense to me

**Pull request merged**